edx_dl/edx_dl.py

```diff
@@ -715,7 +716,12 @@ def download_url(url, filename, headers, args):
         # order) is due to different behaviors in different Python versions
         # (e.g., 2.7 vs. 3.4).
         try:
-            urlretrieve(url, filename)
+            if 'zip' in url and 'mitxpro' in url:
+                urlretrieve(url, filename)
+            else:
+                r = requests.get(url, headers=headers)
+                with open(filename, 'wb') as fp:
+                    fp.write(r.content)
```

Comment:
Can't both of these `if` branches be unified? Unless I'm missing something, I believe that simply using requests could do the job...

Reply:
The compressed zip files could not be downloaded with requests module. I was only able to download them via urlretrieve function. This fix is only for the mitxpro course zip files.